packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+
+    for (const row of csvData) {
+      if ((row["variant.prices"] as object[]).length) {
+        await this.handleVariantPrices(row, regionRepo)
+      }
+
+      if (row["variant.id"]) {
+        variantsUpdate.push(row)
+      } else {
+        variantsCreate.push(row)
+      }
+
+      // save only first occurrence
+      if (!seenProducts[row["product.handle"] as string]) {
+        row["product.profile_id"] = shippingProfile
+        ;(row["product.product.id"] ? productsUpdate : productsCreate).push(row)
+
+        seenProducts[row["product.handle"] as string] = true
+      }
+    }
+
+    return {
+      [OperationType.ProductCreate]: productsCreate,
+      [OperationType.VariantCreate]: variantsCreate,
+      [OperationType.ProductUpdate]: productsUpdate,
+      [OperationType.VariantUpdate]: variantsUpdate,
+    }
+  }
+
+  /**
+   * Prepare prices records for insert - find and append region ids to records that contain a region name.
+   *
+   * @param row - An object containing parsed row data.
+   * @param regionRepo - Region repository.
+   */
+  protected async handleVariantPrices(
+    row,
+    regionRepo: RegionRepository
+  ): Promise<void> {
+    const prices: Record<string, string | number>[] = []
+
+    for (const p of row["variant.prices"]) {
+      const record: Record<string, string | number> = {
+        amount: p.amount,
+      }
+
+      if (p.regionName) {
+        const region = await regionRepo.findOne({
+          where: { name: p.regionName },
+        })
+
+        if (!region) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Trying to set a price for a region ${p.regionName} that doesn't exist`
+          )
+        }
+
+        record.region_id = region!.id
+      } else {
+        record.currency_code = p.currency_code
+      }
+
+      prices.push(record)
+    }
+
+    row["variant.prices"] = prices
+  }
+
+  /**
+   * A worker method called after a batch job has been created.
+   * The method parses a CSV file, generates sets of instructions
+   * for processing and stores these instructions to Redis.
+   *
+   * @param batchJobId . An id of a job that is being preprocessed.
+   */
+  async preProcessBatchJob(batchJobId: string): Promise<void> {
+    const batchJob = await this.batchJobService_.retrieve(batchJobId)
+
+    const csvFileKey = (batchJob.context as ImportJobContext).fileKey
+    const csvStream = await this.fileService_.getDownloadStream({
+      fileKey: csvFileKey,
+    })
+
+    const data = await this.csvParser_.parse(csvStream)
+    const results = await this.csvParser_.buildData(data)
+
+    const ops = await this.getImportInstructions(results)
+
+    await this.setImportDataToRedis(batchJobId, ops)
+
+    await this.batchJobService_.update(batchJobId, {
+      context: {
+        // number of update/create operations to execute
+        total: Object.keys(ops).reduce((acc, k) => acc + ops[k].length, 0),
```

Comment:
Todo: we should put some insight to 
result under stat_descriptors for the user in case of dry_run so that he knows what will happen. You can see an example in the export